force-app/main/default/classes/CourseHandler.cls

```diff
@@ -58,4 +58,24 @@ global with sharing class CourseHandler {
         RestContext.response.addHeader('Content-Type', 'application/json');
         RestContext.response.responseBody = Blob.valueOf(JSON.serialize(wrapperList));
     }
+
+    public override void onBeforeUpdate(Map<Id, sObject> triggerOldMap) {
+        List<Course__c> canceledCourses = new List<Course__c>();
+        for (Course__c course : (List<Course__c>) records) {
+            if (course.get('Cancel__c') != false) {
+                canceledCourses.add(course);
+            }
+        }
+        if (!canceledCourses.isEmpty()) {
+            changeCanceledCourseTitle(canceledCourses);
+        }
+    }
+
+    public static void changeCanceledCourseTitle(
+        List<Course__c> canceledCourses
+    ) {
+        for (Course__c course : canceledCourses) {
+            course.Name = 'AVLYST ' + course.Name;
+        }
+    }
```

Comment:
Trigger-logikken bør ikke ligge i samme klasse som API-et. Det bør ligge i en egen klasse, se klassene Contact_Create og Course_AddUrl for eksempler. Rammeverket for triggere krever også at det opprettes custom metadata for å koble sammen triggeren og klassen som skal kjøre.